cr-examples/onnx/src/test/java/oracle/code/onnx/MNISTDemo.java

```diff
@@ -147,18 +148,20 @@ public void keyPressed(KeyEvent e) {
                     scaledImage.getData().getSamples(0, 0, IMAGE_SIZE, IMAGE_SIZE, 0, imageData);
                     var imageTensor = Tensor.ofShape(new long[]{1, 1, IMAGE_SIZE, IMAGE_SIZE}, imageData);
 
-                    var prediction = OnnxRuntime.execute(MethodHandles.lookup(),
-                            () -> cnn(imageTensor));
-
-                    var result = prediction.data().toArray(ValueLayout.JAVA_FLOAT);
-                    var report = new StringBuilder("<html>");
-                    for (int i = 0; i < result.length; i++) {
-                        var w = result[i];
-                        report.append("&nbsp;<font size=\"%d\" color=\"#%s\">%d</font>&nbsp;(%.1f%%)&nbsp;<br><br><br>"
-                                .formatted((int)(20 * w) + 3, COLORS[(int)(5.99 * w)], i, 100 * w));
+                    try (Arena onnxSession = Arena.ofConfined()) {
```

Comment:
Please don't call it a session. Onnx session is created below during the execution.

Reply:
I can surely change the name -- but note that the MNISTDemo code does NOT create an onnx session explicitly -- not does it care whether a session or an env is constructed under the hood.

I believe the question here is what kind of high-level API would it make sense for babylon users to interact with -- which is a separate question from "how should Java bindings for ONNX look like". It seems we're going in a direction (not just in this PR, but in the last 2-3) where we keep hiding more and more of the ONNX stuff from the user facing code, and we just want to provide a simple abstraction to execute a model (expressed as a reflectable method).

Reply:
> I believe the question here is what kind of high-level API would it make sense for babylon users to interact with -- which is a separate question from "how should Java bindings for ONNX look like"

(at least that's how I interpreted @PaulSandoz comment: https://github.com/openjdk/babylon/pull/332#issuecomment-2691545090)